index.js

```diff
@@ -73,6 +75,12 @@ function gulpFlatBlog(templates) {
     var file = new File({
       path: 'index.html'
     });
+    for (i in posts) {
```

Comment:
You mentioned running into duplicated posts and added this check. The duplicates are a result of the location of the `pageIndex` variable, which is outside of the `gulpFlatBlog` function. In theory, if someone had 2 configurations for `gulpFlatBlog` in their gulpfile (for example, blogs in separate directories), this means posts from both would be accumulated in the `pageIndex` array. The existing `posts` variable is scoped within the `gulpFlatBlog` function, but still accessible from both `singlePost` and `allPosts`.